polkadot/node/subsystem-types/src/messages.rs

```diff
@@ -204,6 +204,50 @@ pub enum CandidateValidationMessage {
 	},
 }
 
+/// Extends primitives::PvfExecKind to have a separate value for duspute requests
+/// which is important for prioritization.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum PvfExecPriority {
+	/// For dispute requests
+	Dispute,
+	/// For approval requests
+	Approval,
+	/// For backing requests from system parachains.
+	BackingSystem,
+	/// For backing requests.
+	Backing,
+}
+
+impl From<PvfExecPriority> for PvfExecKind {
+	fn from(exec: PvfExecPriority) -> Self {
+		match exec {
+			PvfExecPriority::Dispute => PvfExecKind::Approval,
+			PvfExecPriority::Approval => PvfExecKind::Approval,
+			PvfExecPriority::BackingSystem => PvfExecKind::Backing,
+			PvfExecPriority::Backing => PvfExecKind::Backing,
+		}
+	}
+}
+
+impl PvfExecPriority {
+	/// Returns an iterator over the variants of `PvfExecPriorityKind` in order
+	pub fn iter() -> impl Iterator<Item = PvfExecPriority> {
+		[Self::Dispute, Self::Approval, Self::BackingSystem, Self::Backing]
```

Comment:
This won't age when you add new variants people will forget to add them here, you can use crates for this like: https://docs.rs/strum/latest/strum/derive.EnumIter.html.